### PR TITLE
[WIP] Signal Scheme

### PR DESCRIPTION
**Pull request by mrrobot16**

### CODE REVIEW

**Comment by plsdlr** [comment]
WIP Spezification (for Discussion):

IPFS Proposal:

`{
      description: 'Setting new header Image',
      title: 'New Header Image',
      url: 'http://swift.org/modest',
      key: 'Header',
      value: 'https://de.wikipedia.org/wiki/Wald#/media/Datei:Laurisilva_en_el_Cubo_de_la_Galga.jpg',
    }`

Graph Endpoint:


`  signals:[
         {
           data:
             '{"Header":"https://de.wikipedia.org/wiki/Wald#/media/Datei:Laurisilva_en_el_Cubo_de_la_Galga.jpg"}',
           id:
             '0x86e9fe552e75e4fc51f46e4efc128628ecd5ada7'
         }
       ]
    }`

File: test/0.0.1-rc.16/UGenericSchemeSignal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,287 @@
+import {
```
Comment (orenyodfat) [inline test/0.0.1-rc.16/UGenericSchemeSignal.spec.ts]: UGenericScheme is deprecated . please use GenericScheme

File: src/mappings/UGenericScheme/mapping.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { store } from '@graphprotocol/graph-ts';
```
Comment (orenyodfat) [inline src/mappings/UGenericScheme/mapping.ts]: UGenericScheme is deprecated . please use GenericScheme

File: src/domain/signal.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
+import { Bytes, ipfs, json, JSONValueKind, store } from '@graphprotocol/graph-ts';
+import { Signal, Proposal } from '../types/schema';
+import { debug } from '../utils';
+
+export function getSignal(id: string): Signal {
+  let sig = store.get('Signal', id) as Signal;
```
Comment (orenyodfat) [inline src/domain/signal.ts]: how about Signal.load(id) ?

File: src/domain/signal.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+function saveSignal(signal: Signal): void {
+  signal.save()
+}
+
+export function writesignal(signalId: string, proposalId: string): void {
+  readProposal(signalId, proposalId)
+}
+
+export function generatestring(key: string, value: string, signal: Signal ): void {
+  const curlybraceopen = String.fromCharCode(123)
```
Comment (orenyodfat) [inline src/domain/signal.ts]: How about `const curlybraceopen = '{' `

File: src/domain/signal.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    if(searchstringposition === -1){
+      appenddict(newkeypair,signal,curlybraceclose,comma)
+    }
+    else{
+      replaceindict(newkeypair,key,signal,curlybraceopen,curlybraceclose,comma)
+  }
+}
+}
+
+function replaceindict(newkeypair: string,
```
Comment (orenyodfat) [inline src/domain/signal.ts]: replaceIndict (camelCase)

File: src/domain/signal.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+function appenddict(keypair: string, signal: Signal, curlybraceclosed:string, comma:string):void{
+  let signalsubstring = signal.data.substring(0,signal.data.length-1)
+  signal.data = signalsubstring.concat(comma).concat(keypair).concat(curlybraceclosed)
+  debug(signal.data);
+  saveSignal(signal);
+}
+
+export function readProposal(id: string, proposalId: string): void {
```
Comment (orenyodfat) [inline src/domain/signal.ts]: how about adding this code at https://github.com/daostack/subgraph/blob/master/src/domain/proposal.ts#L64

**Pull request closed without merging**